src/components/issue-event-list-item.component.js

```diff
@@ -0,0 +1,520 @@
+          <View style={styles.eventTextContainer}>
+            <Text style={{ padding: 3 }}>
+              <ActorLink actor={actor} /> reopened this
+            </Text>
+          </View>
+          <Date date={createdAt} />
+        </View>
+      </View>
+    );
+  }
+}
+
+class Merged extends Component {
+  props: {
+    event: Object,
+  };
+
+  render() {
+    const {
+      actor,
+      commit_id: commitId,
+      created_at: createdAt,
+    } = this.props.event;
+
+    // TODO: should be able to click commit and view changes
+    return (
+      <View style={[styles.container, styles.header]}>
+        <EventIcon
+          name="git-merge"
+          backgroundColor={colors.purple}
+          iconColor={colors.white}
+        />
+        <View style={styles.contentContainer}>
+          <View style={styles.eventTextContainer}>
+            <Text style={{ padding: 3 }}>
+              <ActorLink actor={actor} /> merged{' '}
+              <Bold>{commitId.slice(0, 7)}</Bold>
+            </Text>
+          </View>
+          <Date date={createdAt} />
+        </View>
+      </View>
+    );
+  }
+}
+
+class Renamed extends Component {
+  props: {
+    event: Object,
+  };
+
+  render() {
+    const { actor, rename, created_at: createdAt } = this.props.event;
+
+    return (
+      <View style={[styles.container, styles.header]}>
+        <EventIcon name="pencil" />
+        <View style={styles.contentContainer}>
+          <View style={styles.eventTextContainer}>
+            <Text style={{ padding: 1 }}>
+              <ActorLink actor={actor} /> changed the title from{' '}
+              <Bold>{rename.from.trim()}</Bold> to{' '}
+              <Bold>{rename.to.trim()}</Bold>
+            </Text>
+          </View>
+          <Date date={createdAt} />
+        </View>
+      </View>
+    );
+  }
+}
+
+class Assigned extends Component {
+  props: {
+    event: Object,
+    unassigned: boolean,
+  };
+
+  render() {
+    const { assigner, assignee, created_at: createdAt } = this.props.event;
+    const action = this.props.unassigned ? 'unassigned' : 'assigned';
+
+    return (
+      <View style={[styles.container, styles.header]}>
+        <EventIcon name="person" />
+        <View style={styles.contentContainer}>
+          <View style={styles.eventTextContainer}>
+            <Text style={{ padding: 3 }}>
+              <ActorLink actor={assigner} /> {action}{' '}
+              <ActorLink actor={assignee} />
+            </Text>
+          </View>
+          <Date date={createdAt} />
+        </View>
+      </View>
+    );
+  }
+}
+
+class Milestoned extends Component {
+  props: {
+    event: Object,
+    demilestoned: boolean,
+  };
+
+  render() {
+    const { actor, milestone, created_at: createdAt } = this.props.event;
+    const action = this.props.demilestoned
+      ? 'removed this from'
+      : 'added this to';
+
+    return (
+      <View style={[styles.container, styles.header]}>
```

Comment:
Do you think we need to have all of these separate components for each specific use case? Will it be possible to have 2 or 3 and then with props, be able to extend their `icon`, `user` and `descriptions` for example?

Reply:
yeah, i see exactly what you're saying. i was handling `actor`s poorly at first, so i wasn't quite sure what the more abstract component would look like yet. it makes sense now tho

Reply:
No worries, always easier to start laying down and then refactoring it slightly :)

Cheers, otherwise I can't see much I would change. This is really coming together nicely mate.

Reply:
Oh, and don't forget to add yourself as a contributor with `yarn contributors:add` :)